hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -422,9 +422,23 @@ private void submitConnectToClusterTask() {
                             logger.warning("No connection to cluster: " + clusterId);
                         }
 
-                        submitConnectToClusterTask();
+                        submitConnectToClusterTask(initialConnection);
                     }
                 }
+                if (initialConnection && !activeConnections.isEmpty()) {
+                    // Send state to cluster for async clients in async way.
+                    executor.submit(() -> {
```

Comment:
we are already in an executor here, i believe we don't need to submit it to the executor again